clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -428,6 +433,12 @@ synchronized boolean isReady() {
         return isTransactional() && currentState == State.READY;
     }
 
+    private synchronized boolean wouldTransactionContainPartition(TopicPartition tp) {
```

Comment:
nit: this name is a bit awkward. Could we split this into two checks `isPartitionAdded` and `isPartitionPending` or something like that?